zerver/tests/test_auth_backends.py

```diff
@@ -1652,8 +1704,49 @@ def register_extra_endpoints(self, requests_mock: responses.RequestsMock,
 
         self.email_data = email_data
 
-    def get_account_data_dict(self, email: str, name: str) -> Dict[str, Any]:
-        return dict(email=email, name=name)
+    def get_account_data_dict(self, email: str, name: str, user_avatar_url: str='') -> Dict[str, Any]:
+        return dict(email=email, name=name, avatar_url=user_avatar_url)
+
+    @override_settings(TERMS_OF_SERVICE=None)
+    def test_register_new_invited_user_and_use_github_avatar(self) -> None:
+        """
+        This tests the registration flow in the case where an invitation for the
+        user was generated and it wants to use their Github avatar.
+        """
+        email = 'newuser@zulip.com'
+        name = 'Full Name'
+        subdomain = 'zulip'
+        realm = get_realm('zulip')
+        mock_user_avatar_url = 'https://github.com/user/mock_avatar_url/image.jpeg'
+
+        iago = self.example_user('iago')
+        do_invite_users(iago, [email], [])
```

Comment:
is an invitation necessary here?

Reply:
yes, because otherwise, it doesn't cover this block:

```python
try:
            prereg_user = PreregistrationUser.objects.filter(
                email__iexact=email, realm=realm).latest("invited_at")

            # password_required and full_name data passed here as argument should take precedence
            # over the defaults with which the existing PreregistrationUser that we've just fetched
            # was created.
            prereg_user.password_required = password_required
            update_fields = ["password_required"]
            if full_name:
                prereg_user.full_name = full_name
                prereg_user.full_name_validated = full_name_validated
                update_fields.extend(["full_name", "full_name_validated"])

            if user_avatar_url:
                prereg_user.user_avatar_url = user_avatar_url
                update_fields.extend(['user_avatar_url'])
```
If a user has an invitation, a `PreregistrationUser` is already created, and when this user try to authenticate, I should fill the `user_avatar_url` of that `prereg_user`.

Reply:
Ah, makes sense!

Reply:
I suppose one more test that would be useful is with the avatar being passed in, but `user_avatar_url = False`.

Reply:
Yes!